src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSProxyContext.cs

```diff
@@ -386,6 +386,9 @@ public unsafe void ReleasePromiseHolder(nint holderGCHandle)
                     holder.IsDisposed = true;
                     handle.Free();
                 }
+#if FEATURE_WASM_MANAGED_THREADS
+                Marshal.FreeHGlobal((IntPtr)holder.State);
```

Comment:
We should zero out holder.State after freeing it so any attempts to dereference it after this are null pointer exceptions instead of a use-after-free

Reply:
though without a programmatic null check (which C# will have, but C/jiterp won't) it will just read/write the zero page instead of throw... :/